projects/js-packages/components/components/zendesk-chat/zendesk-chat-script.tsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+import { chatKey } from './constants';
+import { ZendeskChatScriptType } from './types';
+
+const ZendeskChatScript: ZendeskChatScriptType = () => {
+	useEffect( () => {
+		const script = document.createElement( 'script' );
+		const container = document.getElementById( 'zendesk-chat-container' );
+
+		script.src = 'https://static.zdassets.com/ekr/snippet.js?key=' + encodeURIComponent( chatKey );
+		script.type = 'text/javascript';
+		script.id = 'ze-snippet';
+
+		if ( container ) {
+			container.appendChild( script );
```

Comment:
How does the zendesk script work? Does it expect that the chat container exists before the script is added/ run? Or could the chat container be added later on? Is the container here just an anchor for the script to be added into?

Something about assembling the `<script>` element in React feels like an anti-pattern. Would there be a workable way to load the script via `wp_enqueue_script`? I don't think this is an invalid approach - but curious if we can load the script from the backend? Maybe not?

Reply:
For some context, this implementation was to somewhat emulate the [implementation on Calypso](https://github.com/Automattic/wp-calypso/tree/trunk/client/components/presales-zendesk-chat), and while I think there is some value in having similar implementations across the two for ease of updating, I agree that if we can get it to work on the backend with a simple `wp_enque_script`, it may be a better way to go about it

I did not think of doing it this way, so I have not confirmed that it will/will not work. In Claypso they use a function called `loadScript` which doesn't really exist, that I know of or could find, in this codebase, and it is a semi-complicated function that I didn't just want to throw into this PR. 

I'll go ahead and see if loading it from the backend can meet our needs if you still think that's the better approach after laying out the context

Reply:
> Is the container here just an anchor for the script to be added into?

It is my understanding that the container is just an anchor for the script to be added into

Reply:
I see how this might seem like an anti-pattern - and I'd prefer to have it loaded in the backend as well - but this is actually a common way to embed external scripts into React components. Considering what y'all have discussed, I also think we should go with this solution.